pkg/koordlet/runtimehooks/reconciler/reconciler.go

```diff
@@ -385,6 +445,34 @@ func (c *reconciler) reconcilePodCgroup(stopCh <-chan struct{}) {
 					}
 				}
 			}
+
+			for _, r := range globalCgroupReconcilers.allPodsLevel {
+				var reconcileFn reconcileFunc4AllPods
+				currentPods := make([]protocol.HooksProtocol, 0)
+				for _, podMeta := range podsMeta {
+					if fn, ok := r.fn4AllPods[r.filter.Filter(podMeta)]; ok {
+						podCtx := protocol.HooksProtocolBuilder.Pod(podMeta)
+						currentPods = append(currentPods, podCtx)
+						if reconcileFn == nil {
+							reconcileFn = fn
```

Comment:
It seems the last reconcileFn would be picked. Why not multiple funcs?

Reply:
because there only exist one reconcileFn for each item in globalCgroupReconcilers.allPodsLevel. Here we want to get the reconcileFn and all pods.
```
allPodsLevel:          map[string]*cgroupReconciler{},
...
fn4AllPods  map[string]reconcileFunc4AllPods
```

Reply:
The definition above seems to map the reconcileFunc4AllPods by the condition type. So it could be multiple reconcileFns for all condition types. If the reconcileFns is multiple, the next pod should not rely on the condition type of the previous pods. If the reconcileFns is single, the reconcileFn can be retrieved outside the pod loop.

Reply:
Yes, you're right. There could be muliple reconcileFns.